### ISSUE TITLE
Group Norm crashes on Apple M1/MPS devices for versions 2.0+

### ISSUE THREAD

**Issue by KE7** [pytorch/pytorch#99981]
### 🐛 Describe the bug

I ran the following notebook locally under 3 versions of PyTorch (1.13.1, 2.0.0, and local build of main 2.1.0a0+git466adab) https://colab.research.google.com/drive/1gxdkgRVfM55zihY9TFLja97cSVZOZq2B?usp=sharing 

I isolated the error to the group norm layer (nn.GroupNorm(n_groups, out_channels)). With a group norm (regardless of if the network is on cpu or sent to mps) the very first loss.backwards() causing the python kernel to crash. If you swap this layer out with a batch norm or instance norm, training proceeds as normal

### Versions

Collecting environment information...
PyTorch version: 2.1.0a0+git466adab
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 13.3.1 (arm64)
GCC version: Could not collect
Clang version: 14.0.3 (clang-1403.0.22.14.1)
CMake version: version 3.22.1
Libc version: N/A

Python version: 3.8.16 (default, Mar  1 2023, 21:18:45)  [Clang 14.0.6 ] (64-bit runtime)
Python platform: macOS-13.3.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M1 Max

Versions of relevant libraries:
[pip3] numpy==1.23.5
[pip3] torch==2.1.0a0+git466adab
[conda] numpy                     1.23.5                   pypi_0    pypi
[conda] torch                     2.1.0a0+git466adab           dev_0    <develop>

cc @kulinseth @albanD @malfet @DenisVieriu97 @razarmehr @abhudev

**Comment by drisspg** [comment]
Could you create a smaller reproduction script?

**Comment by KE7** [comment]
That’s not my research paper so I will try to slim down their collab notebook. 

@qqaatw It is a PyTorch issue because if you take the same exact notebook and run it in PyTorch 1.13.1 it works just fine. With 2.0, or nightly the issue is observed

**Comment by qqaatw** [comment]
OK, so it sounds like a regression and not so much related to mps.

**Comment by KE7** [comment]
It’s a regression that only happens on MPS.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by qqaatw** [comment]
> With a group norm (regardless of if the network is on cpu or sent to mps) the very first loss.backwards() causing the python kernel to crash.

I am a bit confused?

**Comment by KE7** [comment]
If you review the notebook, you can see that there is a MSE loss that is computing in a training loop. The backwards call errors out immediately.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by KE7** [comment]
@qqaatw Try this collab link that I slimmed down. I also cut down the 1 unet to 1 hidden layer. Download the notebook and just run all the steps to reproduce
